pandas/plotting/_core.py

```diff
@@ -939,15 +939,15 @@ def __call__(self, *args, **kwargs):
                         f"{kind} requires either y column or 'subplots=True'"
                     )
                 if y is not None:
-                    if is_integer(y) and not data.columns.holds_integer():
+                    if is_integer(y) and data.columns._should_fallback_to_positional:
```

Comment:
i dont think these are exactly the same.  _should_fallback_to_positional is overriden by Index subclasses

Reply:
Hmm, my instinct is that the old code easily give wrong result by giving an integer to an index that has `_should_fallback_to_positional` set to False and it should instead have been crystal clear if x/y are meant for .loc/.iloc instead of this ix. but this is difficult to prove and I'm not very familiar with the plotting code, so I've changed to use `_holds_integer`.

Reply:
i agree the current code is sketchy.  hopefully we'll figure out what the original intent was and can change it to something more like `is_integer_dtype(data.columns.dtype)`